src/native/eventpipe/ep-session.c

```diff
@@ -511,6 +627,64 @@ ep_session_write_all_buffers_to_file (EventPipeSession *session, bool *events_wr
 	return !ep_file_has_errors (session->file);
 }
 
+bool
+ep_tracepoint_write (
+	EventPipeSession *session,
+	ep_rt_thread_handle_t thread,
+	EventPipeEvent *ep_event,
+	EventPipeEventPayload *payload,
+	const uint8_t *activity_id,
+	const uint8_t *related_activity_id,
+	ep_rt_thread_handle_t event_thread,
+	EventPipeStackContents *stack)
+{
+#ifdef HAVE_SYS_UIO_H
+	EventPipeProvider *provider = ep_event_get_provider (ep_event);
+
+	EventPipeSessionProviderList *session_provider_list = ep_session_get_providers (session);
+	EventPipeSessionProvider *session_provider = ep_session_provider_list_find_by_name (ep_session_provider_list_get_providers (session_provider_list), ep_provider_get_provider_name (provider));
+
+	uint32_t event_id = ep_event_get_event_id (ep_event);
+
+	dn_umap_t *event_id_to_tracepoint_map = ep_session_provider_get_event_id_to_tracepoint_map (session_provider);
+	dn_umap_it_t found1 = dn_umap_find (event_id_to_tracepoint_map, &event_id);
+	EventPipeTracepoint *tracepoint = dn_umap_it_value_t (found1, EventPipeTracepoint *);
+	if (tracepoint == NULL) {
+		// If we don't have a tracepoint for this event_id, use the default tracepoint
+		tracepoint = session_provider->default_tracepoint;
+	}
+	if (tracepoint == NULL) {
+		// No tracepoint for this event_id and no default tracepoint, so we can't write the event.
+		return false;
```

Comment:
What does the bool return on this function signify? Its unclear why some of these paths return false and others return true.